provider/cloud.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2017 Giant Swarm.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"io"
+
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+const providerName string = "kvm-operator"
+
+type cloud struct {
+	instances cloudprovider.Instances
+}
+
+func newCloud(config io.Reader) (cloudprovider.Interface, error) {
+	return &cloud{
+		instances: newInstances(),
+	}, nil
+}
+
+func init() {
+	cloudprovider.RegisterCloudProvider(providerName, func(config io.Reader) (cloudprovider.Interface, error) {
+		return newCloud(config)
+	})
```

Comment:
What is this doing? Why are we registering a cloud provider which is basically just fake?

Reply:
+1 I also don't follow this. Are you going to add some functionality here? What will it be? Why not removing provider package now, and add it when it does something useful?